sw/device/lib/dif/dif_plic.c

```diff
@@ -138,7 +137,7 @@ static size_t plic_num_irq_reg(void) {
  * the offset for a specific IRQ source ID (ID 33 would be IE01, ...).
  */
 static ptrdiff_t plic_offset_from_reg0(dif_plic_irq_id_t irq) {
-  uint8_t register_index = PLIC_ID_TO_INDEX(irq) / PLIC_ID_TO_INDEX_REG_SIZE;
+  uint8_t register_index = irq / PLIC_ID_TO_INDEX_REG_SIZE;
```

Comment:
@lenary Here's the part I revised. ID read from CC register is matched to the bit position

Reply:
This looks good. I think you'll also need to change `plic_num_priority_reg` to ensure that `plic_reset` resets the final priority register, but that should be it.

I'd like a second review by @silvestrst of the code, just to make sure I haven't missed anything.